deepspeed/runtime/fp16/onebit/zoadam.py

```diff
@@ -228,6 +228,11 @@ def step(self, closure=None, grads=None):
                                         state['server_error'],
                                         self.deepspeed.local_rank)
                                     exp_avg.mul_(beta1).add_(1 - beta1, grad_onebit)
+                                    if 'exp_avg_mask' in group:
+                                        if grad_onebit.device != group['exp_avg_mask'].device:
+                                            group['exp_avg_mask'] = group['exp_avg_mask'].to(
+                                                device=grad_onebit.device)
+                                        grad_onebit.mul_(group['exp_avg_mask'])
```

Comment:
I think this actually needs to be added before line 230 "exp_avg.mul_(beta1).add_(1 - beta1, grad_onebit)"